scripts/test-claude-stub.py

```
@@ -0,0 +1,879 @@
+    def test_an_interrupt_the_child_cannot_have_seen_is_forwarded(self):
+        self.assertEqual([int(signal.SIGINT)], self._run_with_interrupt(False))
+
+
+class EarlyInterruptTests(SignalTargetsFixture):
+    def test_a_ctrl_c_before_claude_runs_exits_130_and_removes_the_sandbox(self):
+        root = Path(tempfile.mkdtemp(prefix="claudestub-test-"))
+        self.addCleanup(shutil.rmtree, root, ignore_errors=True)
+        created: list[str] = []
+        real_mkdtemp = tempfile.mkdtemp
+
+        def recording_mkdtemp(*args, **kwargs):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `recording_mkdtemp`

(ANN202)